### PR TITLE
Add configFileName templating, to provide additional controls for externally managed configMaps

### PR DESCRIPTION
**Pull request by conallob**
* Add templating to allow users to define their own command line arguments, with reasonable defaults if no customisation is
  defined

* Add templating to allow users to control the mount path for the configMap

### CODE REVIEW

**Comment by conallob** [comment]
I'm still new to helm, so feedback is very much welcome

File: cloudprober/templates/deployment.yaml
Comment (manugarg) [inline cloudprober/templates/deployment.yaml]: We can drop this arg now. It doesn't do anything.

File: cloudprober/templates/deployment.yaml
Comment (manugarg) [inline cloudprober/templates/deployment.yaml]: So I like the idea of extraArgs in #35, i.e. you always have the config_file arg. If we have disable that ever, there should like be another template variable for that.

and, for extraArgs, something like this will likely work better (i.e. explicitly iterate over them instead of indenting the input array.
https://github.com/grafana/helm-charts/blob/4401adc2780fba7a5a8da3dcda1a2d6fb039bccc/charts/grafana/templates/_pod.tpl#L945

File: cloudprober/templates/deployment.yaml
Comment (conallob) [reply cloudprober/templates/deployment.yaml]: Done

File: cloudprober/templates/deployment.yaml
Comment (conallob) [reply cloudprober/templates/deployment.yaml]: I'm ok with extraArgs, but I would like a way to override the name of the config file. A variable to set a config file name suffix would also work for me

**Comment by conallob** [comment]
To give some additional context:

My use case is that I have multiple cloudprober_${environment}.cfg files, since I'm running Cloudprober both in and outside of K8s. While I could grandfather cloudprober.cfg to be the k8s config, I'd prefer to be able to call it cloudprober_k8s.cfg so it's self documenting

File: cloudprober/templates/deployment.yaml
Comment (manugarg) [reply cloudprober/templates/deployment.yaml]: * I was wondering why you'd want to override the config path. This seems entirely internal to the helm chart.
* If you're passing the configMap (i.e. helm is not creating the configMap), config can be named differently. So, adding a variable called 'configFileName' will likely make sense. We can use it in deployment.yaml and here: https://github.com/cloudprober/helm-charts/blob/9363f8ffba44f8c12df11b268e85a029e64e23a1/cloudprober/templates/configmap.yaml#L7
(I am guessing that's the use case you described in the comment?).

**Review by manugarg — APPROVED** [review]

File: cloudprober/templates/deployment.yaml
Comment (conallob) [reply cloudprober/templates/deployment.yaml]: > * I was wondering why you'd want to override the config path. This seems entirely internal to the helm chart.

The config path issue is due to inconsistencies between the config include statement and the file targets. Config includes can take relative paths, but the file based targets require an absolute path. I can workaround this by symlinking /cfg  to /etc/cloudprober in my non k8s environments.

> * If you're passing the configMap (i.e. helm is not creating the configMap), config can be named differently. So, adding a variable called 'configFileName' will likely make sense. We can use it in deployment.yaml and here: https://github.com/cloudprober/helm-charts/blob/9363f8ffba44f8c12df11b268e85a029e64e23a1/cloudprober/templates/configmap.yaml#L7

Correct, this is my problem. I generate my configMap and pass it in, but can never use it because I don't have a cloudprober.cfg file within it.

I've created the configFileName variable as you've suggested and added it to both configmap.yaml and deployment.yaml

**Comment by conallob** [comment]
Friendly ping?

I lost my merge approval when I added configFileName to templates/configmap.yaml

File: cloudprober/templates/configmap.yaml
Comment (manugarg) [inline cloudprober/templates/configmap.yaml]: Just a couple of more things:
- Instead of setting the default here, it's better to just put in values.yaml, like how we do for the config format.
- Please add this variable to values.yaml for documentation and for default value, likely here:
https://github.com/cloudprober/helm-charts/blob/56e1e69585fbf519c0fd6bee9fcbb1468fbaabcb/cloudprober/values.yaml#L23

File: cloudprober/templates/deployment.yaml
Comment (manugarg) [inline cloudprober/templates/deployment.yaml]: See my comment above about keeping default value in values.yaml. Also while here can you just remove --logtostderr.

**Comment by conallob** [comment]
My edits to values.yaml will need additional approval. Feel free to make a maintainer edit before merging, if you disagree with my use of reasonable default values for configFileName

File: cloudprober/values.yaml
Comment (manugarg) [inline cloudprober/values.yaml]:
```suggestion
# map. If you're managing the configMap outside of helm and config file name
# is different, you should set the value of configFileName as well.
```

File: cloudprober/values.yaml
Comment (manugarg) [inline cloudprober/values.yaml]:
```suggestion
# You can override the default config filename using the following field. It's
# typically useful if you're managing the configMap outside of helm.
```

**Review by manugarg — COMMENTED** [review]
Minor comment fixes. I'll commit them.

File: cloudprober/templates/configmap.yaml
Comment (manugarg) [inline cloudprober/templates/configmap.yaml]:
```suggestion
  {{- .Values.configFileName }}.{{- .Values.configFormat }}: |
```

File: cloudprober/templates/deployment.yaml
Comment (manugarg) [inline cloudprober/templates/deployment.yaml]:
```suggestion
          args: ['--config_file', '/cfg/{{- .Values.configFileName -}}.{{- .Values.configFormat -}}']
```

**Review by manugarg — COMMENTED** [review]
Couple of more that I'll commit shortly.

**Comment by manugarg** [comment]
@conallob Charts version 1.13.18 will have the latest changes.

**Pull request merged**
